Compile bug: compile fail

### Git commit

f1bad23f880d059183ac518e9efa20523c7f3fbb

### Operating systems

Mac

### GGML backends

CPU, Metal

### Problem description & steps to reproduce

mkdir build
cd build
cmake ..
make

### First Bad Commit

_No response_

### Compile command

```shell
make
```

### Relevant log output

```shell
:build gxy$ make
[  3%] Built target ggml-base
[  5%] Built target ggml-metal
[ 10%] Built target ggml-cpu
[ 10%] Built target ggml-blas
[ 10%] Built target ggml
[ 10%] Linking CXX shared library ../bin/libllama.dylib
ld: malformed 64-bit a.b.c.d.e version number: 0.0.7055
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [bin/libllama.0.0.7055.dylib] Error 1
make[1]: *** [src/CMakeFiles/llama.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -139,6 +139,7 @@ add_library(llama
 set_target_properties(llama PROPERTIES
     VERSION ${LLAMA_INSTALL_VERSION}
     SOVERSION 0
+    MACHO_CURRENT_VERSION 0 # keep macOS linker from seeing oversized version number
 )
 
 target_include_directories(llama PRIVATE .)

--- a/tools/mtmd/CMakeLists.txt
+++ b/tools/mtmd/CMakeLists.txt
@@ -16,6 +16,7 @@ add_library(mtmd
 set_target_properties(mtmd PROPERTIES
     VERSION ${LLAMA_INSTALL_VERSION}
     SOVERSION 0
+    MACHO_CURRENT_VERSION 0 # keep macOS linker from seeing oversized version number
 )
 
 target_link_libraries     (mtmd PUBLIC ggml llama)

